docs/content/direct/multi-wec-aggregated-status.md

```diff
@@ -0,0 +1,171 @@
+
+Logic:
+- If **enabled** and number of WECs == 1 → use existing singleton mechanism.
+- If **enabled** and number of WECs > 1 → invoke new multi-WEC aggregation logic.
+- If **disabled**, retain existing empty status behavior.
+
+```yaml
+spec:
+  clusterSelectors:
+  - matchLabels: {"location": "edge"}
+  downsync:
+  - objectSelectors:
+    - matchLabels: {"app.kubernetes.io/name": "nginx-multi"}
+    wantMultiWECReportedState: true
+```
+
+The flag is defined under each downsync rule, consistent with [example scenario 4](https://docs.kubestellar.io/release-0.28.0/direct/example-scenarios/#scenario-4-singleton-status).
+
+---
+
+### 2. Introduce `handleMultiReportedState()` Function
+
+A new controller function will handle multi-WEC aggregation:
+
+**Responsibilities:**
+1. Collect `.status` from all target WECs using existing status collection logic (see [pkg/status/singletonstatus.go#L64](https://github.com/kubestellar/kubestellar/blob/main/pkg/status/singletonstatus.go#L64)).  
+2. Aggregate status data according to defined field rules (numeric, conditions, timestamps, string) or use a per-kind aggregator if it is a known KIND.  
+3. Update the aggregated `.status` back into the WDS workload object.
+
+**Focus:** Aggregation of fields relevant to ArgoCD health interpretation.
+
+---
+
+
+### 3. Status Aggregation Rules
+
+The aggregation logic depends on the **workload object kind**.
+
+#### Case 1 – Known Workload Kinds
+If the workload kind is **recognized by ArgoCD** (for example Deployment, StatefulSet, DaemonSet, ReplicaSet), the controller applies **predetermined field aggregation rules** consistent with ArgoCD’s native health evaluation.  
+
+Examples:
+- **Deployment:** mark `Available=True` when `availableReplicas == replicas`.
+- **StatefulSet:** mark `Ready=True` when all replicas are `current` and `ready`.
+- **DaemonSet:** mark `Ready=True` when `numberAvailable == desiredNumberScheduled`.
+
+These mappings are defined statically in the controller. No per-field data type aggregation is performed in this case.
+
+#### Case 2 – Unknown Workload Kinds
+If the workload kind is **not known to ArgoCD**, the controller performs **generic per-field aggregation** to derive a combined `.status`.
+
+| Field Type | Aggregation Logic | Description |
+|-------------|------------------|--------------|
+| **Numeric** | Average or Minimum | For counts like `replicas` or `availableReplicas`. |
+| **Condition** | Group by `type`; aggregate per condition type | Use AND for Available/Ready, OR for Progressing/Degraded. |
+| **Timestamp** | Latest timestamp | Reflects the most recent cluster update. |
+| **String** | Latest value | Keeps newest message or reason from clusters. |
+
+This two-step approach keeps aggregation consistent with ArgoCD’s semantics for known workloads while providing a fallback for unknown kinds.
+
+---
+
+### 4. Implementation Approach
+
+1. Extend BindingPolicy CRD with `wantMultiWECReportedState`.
```

Comment:
The `Binding` type needs the same extension, since a BindingPolicy gets resolved to a Binding.

Really, it is DownsyncModulation that needs extending (that is where the singleton request is); both Binding and BindingPolicy use DownsyncModulation.